### ISSUE TITLE
Context-root attribute for server.xml web-ext element ignored

### ISSUE THREAD

**Issue by aguibert** [OpenLiberty/open-liberty#696]
In Liberty it is possible to define binding attributes via server.xml.  According to our documentation, if we have an enterprise application (.ear) named "earApp" which contains a web archive (.war) named "warModule", we should be able to re-map the default context root of the warModule to a different context root using the following server.xml config:

```xml
<enterpriseApplication location="earApp.ear">
    <web-ext moduleName="warModule" context-root="somethingElse"/>
</enterpriseApplication>
```

If we use the following configuration, the context root for the web module will be `/warModule` but it should be `/somethingElse`.

**Comment by brenthdaniel** [comment]
There's a lot going on here.. 

1) The scenario mentioned should never have been supported or documented. Context root in the ibm-web-ext schema is a sub element of web-ext, so the actual configuration should be:
<web-ext moduleName="warModule">
  <context-root uri="somethingElse"/>
</web-ext>

This is caused by a bug in the code generation from dd models. We don't correctly handle the case where a subelement with one attribute is turned into an attribute in the models. This affects 11 elements in ibm-web-ext as well as shared-session-context from ApplicationExt. 

This will need to be fixed so that we can support copy-paste of ibm-web-ext into server.xml, but we'll also have to continue to support what we have now to maintain backward compatibility. 

2) The documented form of specifying context-root does not work when a WAR is inside an EAR. The other incorrectly documented elements should be fine. The web container is trying to access the context root before the container has been completely initialized. That's fine for loading ibm-web-ext.xml, but doesn't work for config overrides.

**Comment by brenthdaniel** [comment]
Problem 2 was fixed by @ewhoch in #1553 

Problem 1 we have decided to live with. We will only support the <web-ext contextRoot="someRoot"/> form. The documentation is being updated to clarify that these specific cases diverge from the old app binding and extension models.
